src/prototype_creep_resources.js

```diff
@@ -12,17 +23,10 @@ Creep.prototype.harvesterBeforeStorage = function() {
   methods.push(Creep.transferEnergy);
 
   let structures = this.room.find(FIND_MY_CONSTRUCTION_SITES, {
-    filter: function(object) {
-      if (object.structureType == STRUCTURE_RAMPART) {
-        return false;
-      }
-      if (object.structureType == STRUCTURE_WALL) {
-        return false;
-      }
-      if (object.structureType == STRUCTURE_CONTROLLER) {
-        return false;
-      }
-      return true;
+    filter: (s) => {
+      return (s.structureType != STRUCTURE_RAMPART &&
+        s.structureType != STRUCTURE_WALL &&
+        s.structureType != STRUCTURE_CONTROLLER);
```

Comment:
I like simplifying it.

@Nevrdid started to introduce `tables` for that. I started an issue #36 maybe we could discuss and agree on a common solution.